### PR TITLE
T20: Display Individual digits

### PR DESCRIPTION
**Pull request by Ashish02112001**

### CODE REVIEW

File: Program.cs [hunk 1]
```diff
@@ -1,7 +1,46 @@
-﻿namespace Training {
+﻿// --------------------------------------------------------------------------------------------
+// Training ~ A training program for new joinees at Metamation, Batch - July 2023.
+// Copyright (c) Metamation India.                                              
+// ------------------------------------------------------------------------
+// Program.cs
+// DISPLAY THE INDIVIDUAL DIGITS OF A GIVEN NUMBER 
+// Displays the individual digits of a given number, which should also handle decimal digits.
+// For example, 355.56 is the input number, output = integral part: 3 5 5; factorial part: 5 6.
```
Comment (asfar-nazar) [inline Program.cs:8]:
```suggestion
// For example, 355.56 is the input number, output = integral part: 3 5 5; fractional part: 5 6.

```

File: Program.cs [hunk 2]
```diff
@@ -1,7 +1,46 @@
...
+               Console.Write ("\nFractional Part: ");
+               PrintDigits (Math.Abs(fractPartx));
+               break;
+            } else { Console.Write ("Enter a valid number: "); }
+         }
       }
+      /// <summary>Prints the individual digits</summary>
+      /// <param name="num">Number from the user</param>
+      static void PrintDigits(int num) {
+         while (num > 0) {
```
Comment (asfar-nazar) [inline Program.cs:37]: You could just convert the number to a string and print individual characters

File: Program.cs [hunk 3]
```diff
@@ -1,7 +1,46 @@
...
       static void Main (string[] args) {
-         Console.WriteLine ("Hello, World!");
+         Console.Write ("Enter a number: ");
+         for (; ; ) {
+            if (decimal.TryParse (Console.ReadLine (), out var num)) {
+               int intPart = (int)num;
+               decimal fractPart = num - intPart;
+               int fractLen = fractPart.ToString ().Length;
+               fractLen = fractPart < 0 ? fractLen - 3 : fractLen - 2;
+               int fractPartx = (int) ((decimal)Math.Pow (10, fractLen) * fractPart);
```
Comment (asfar-nazar) [inline Program.cs:25]: With the number converted to a string, you could just use substring to get the relevant part, instead of doing this computation

**Review by asfar-nazar — CHANGES_REQUESTED** [review]

File: Program.cs [hunk 4]
```diff
@@ -1,7 +1,37 @@
...
+      #region Methods ---------------------------------------------
+      /// <summary>This Method gets input number from the user and prints the individual digits</summary>
+      /// <param name="args">arguments</param>
       static void Main (string[] args) {
-         Console.WriteLine ("Hello, World!");
+         Console.Write ("Enter a number: ");
+         for (; ; ) {
+            if (decimal.TryParse (Console.ReadLine (), out var num)) {
+               string numStr = num.ToString ();
+               string intPart = numStr.Contains ('.') ? numStr.Substring (0, numStr.IndexOf ('.')) : numStr;
```
Comment (asfar-nazar) [inline Program.cs:22]: You are checking if numStr contains '.' twice when you don't need to check it at all. Hint: string 'split'

**Review by asfar-nazar — CHANGES_REQUESTED** [review]

File: Program.cs [hunk 5]
```diff
@@ -1,7 +1,36 @@
...
+         for (; ; ) {
+            if (decimal.TryParse (Console.ReadLine (), out var num)) {
+               string numStr = num.ToString ();
+               string[] digits = numStr.Split ('.');
+               Console.Write ("Integral part: ");
+               foreach (char iP in digits[0]) Console.Write ($"{iP} ");
+               if (digits.Length == 2) {
+                  Console.Write ("\nFractional part: ");
+                  foreach (char fP in digits[1]) Console.Write ($"{fP} ");
+               }break;
```
Comment (asfar-nazar) [inline Program.cs:28]:
```suggestion
               } break;

```

**Review by asfar-nazar — CHANGES_REQUESTED** [review]
What will the output be if user enters -29 as input. Should the output be - 2 9 or just 2 9 ?

**Comment by Ashish02112001** [comment]
> What will the output be if user enters -29 as input. Should the output be - 2 9 or just 2 9 ?

Integral part includes the sign also na so i included.

**Review by asfar-nazar — APPROVED** [review]

**Pull request closed without merging**